test/spmd/test_xla_sharding.py

```diff
@@ -1118,6 +1118,39 @@ def test_mark_manual_sharding(self):
     # The following exception cannot be caught somehow.
     # xt.global_tensor.cpu()
 
+  def test_spmd_full_to_shard_shape(self):
+    x = torch.zeros(8, 4).to(xm.xla_device())
+    with self.assertRaises(RuntimeError):
+      x = torch_xla._XLAC._spmd_full_to_shard_shape(x)
+
+    # Sharded shape
+    xt = xs.mark_sharding(x, self._get_mesh((1, self.n_devices)), (0, 1))
+    xx = torch_xla._XLAC._spmd_full_to_shard_shape(xt.global_tensor)
+
+    hlo = torch_xla._XLAC._get_xla_tensors_hlo([xx])
+    self.assertEqual(xx.shape, (8, 4 // self.n_devices))
+    self.assertIn(f'%custom-call.2 = f32[8,{4//self.n_devices}]{{1,0}}', hlo)
+    self.assertIn(
+        f'custom_call_target="SPMDFullToShardShape", sharding={{manual}}', hlo)
+    self.assertEqual(torch_xla._XLAC._get_xla_sharding_spec(xx), "{manual}")
+
+    # It looks like XLA does't like only having manual sharding in the HLO.
+    # It needs to be paired with SPMDFullToShardShape/SPMDShardToFullShape.
+    # The following exception cannot be caught somehow.
+    # xx.cpu()
```

Comment:
do you intend to keep this `xx.cpu`?

Reply:
Yea, it's more like a note that this won't work... I was trying to use with self.assertRaises but that doesn't capture the exception... I have noticed this before too. When libtpu crashed, it's hard to catch it in the py level. Not sure why. Maybe you have some better ideas?

Reply:
oh I think I run into similar issue before.. The way I handle it was ugly through https://github.com/pytorch/xla/blob/a7a1357d5ee3b575a7eeba523f74607461eddad1/test/spmd/test_dynamo_spmd.py#L172-L181

Reply:
C++ crash on pt level can be caught with `self.assertRaise` but not libtpu level.... I'm not sure why... yea, not even with this hack...

Reply:
cc @will-cromar Do you know how to catch libtpu exception on py? Appreciate your insights.

Reply:
I don't think you can. To make a proper runtime error, you have to raise an exception, and Google internal binaries don't generally do that. I wrote about a similar case in https://github.com/pytorch/xla/issues/6700#issuecomment-1986521945

Reply:
Thanks, Will. That makes a lot of sense now.